pennylane/templates/embeddings/iqp.py

```diff
@@ -170,10 +170,10 @@ def __init__(self, features, wires, n_repeats=1, pattern=None, do_queue=True, id
 
         shape = qml.math.shape(features)
 
-        if len(shape) != 1:
+        if len(shape) not in {1, 2}:
             raise ValueError(f"Features must be a one-dimensional tensor; got shape {shape}.")
 
-        n_features = shape[0]
+        n_features = shape[-1]
```

Comment:
Does this also imply a breaking change?

Reply:
No, because previously, the number of dimensions was expected (and asserted) to be one. That is, previously the index `0` and `-1` were equivalent, and we here pick `-1` because it is additionally compatible with broadcasting :)